src/main/java/org/dependencytrack/search/IndexManager.java

```diff
@@ -339,10 +396,106 @@ public static void delete(final IndexType indexType) {
     }
 
     /**
-     * Returns if the index directory exists or not.
-     * @since 3.4.0
+     * Ensure that all lucene indexes are healthy.
+     */
+    public static void ensureIndexesExists() {
+        Arrays.stream(IndexManager.IndexType.values()).forEach(indexType -> {
+            if (!isIndexHealthy(indexType)) {
+                LOGGER.info("(Re)Building index "+indexType.name().toLowerCase());
+                LOGGER.debug("Dispatching event to reindex "+indexType.name().toLowerCase());
+                Event.dispatch(new IndexEvent(IndexEvent.Action.REINDEX, indexType.getClazz()));
+            }
+        });
+    }
+
+    /**
+     * Check that the index exists and is not corrupted
      */
-    public static boolean exists(final IndexType indexType) {
-        return getIndexDirectory(indexType).exists();
+    private static boolean isIndexHealthy(final IndexType indexType) {
+        LOGGER.info("Checking the health of index "+indexType.name());
+        File indexDirectoryFile = getIndexDirectory(indexType);
+        LOGGER.debug("Checking FS directory "+indexDirectoryFile.toPath());
+        if(!indexDirectoryFile.exists()) {
+            LOGGER.warn("The index "+indexType.name()+" does not exist");
+            return false;
+        }
+        LOGGER.debug("Checking lucene index health");
+        Directory luceneIndexDirectory = null;
+        CheckIndex checkIndex = null;
+        try {
+            File writeLock = Path.of(indexDirectoryFile.getAbsolutePath(), IndexWriter.WRITE_LOCK_NAME).toFile();
+            if (writeLock.exists()) {
+                LOGGER.debug("Stale lock file detected. Deleting it");
+                writeLock.delete();
+            }
+            luceneIndexDirectory = FSDirectory.open(indexDirectoryFile.toPath());
+            checkIndex = new CheckIndex(luceneIndexDirectory);
+            checkIndex.setFailFast(true);
+            checkIndex.setInfoStream(System.out);
```

Comment:
This should be wrapped in a `LOGGER.isDebugEnabled()` condition, as it's producing a lot of noise. When debug logging is not enabled, we could use [`NullPrintStream`](https://commons.apache.org/proper/commons-io/javadocs/api-release/org/apache/commons/io/output/NullPrintStream.html) to discard the output.

Reply:
Done